neqo-http3/src/features/extended_connect/webtransport_session.rs

```diff
@@ -415,7 +415,7 @@ impl WebTransportSession {
             let mut dgram_data = Encoder::default();
             dgram_data.encode_varint(self.session_id.as_u64() / 4);
             dgram_data.encode(buf);
-            conn.send_datagram(dgram_data.as_ref(), id)?;
+            conn.send_datagram(dgram_data.into(), id)?;
```

Comment:
`dgram_data` is an `Encoder` wrapping a `Vec<u8>`. Instead of passing a reference (i.e. `dgram_data.as_ref()`) pass the inner `Vec<u8>` (i.e. `dgram_data.into()`) directly